video/input.h

```diff
@@ -70,26 +74,69 @@ class Input
 
     virtual const std::string getCameraName() const
     {
-        return "unknown_camera";
+        return DefaultCameraName;
     }
 
-    virtual bool readFrame(cv::Mat &outFrame)
+    virtual cv::Size getOutputSize() const
     {
-        return false;
+        return cv::Size();
     }
 
-    virtual cv::Size getOutputSize() const
+    virtual bool needsUnwrapping()
     {
-        return cv::Size();
+        // only panoramic cameras are defined with the camera name specified
+        return getCameraName() != DefaultCameraName;
     }
 
+    virtual bool readFrame(cv::Mat &outFrame) = 0;
+
+    virtual void setOutputSize(const cv::Size &outSize)
+    {}
+
 protected:
     /*
      * Default constructor is protected to stop users directly creating objects
      * of this class.
      */
     Input()
```

Comment:
You can also do this explicitly with ``Input() = delete;`` in C++11.

Reply:
:+1: